src/file_pipe_log/mod.rs

```diff
@@ -221,7 +222,51 @@ pub mod debug {
                     }
                 }
             }
-            assert!(reader.next().is_none())
+
```

Comment:
Put the test in engine.rs. You are not supposed to use `Engine` in a component crate.